animeal/src/Flows/Main/Modules/Home/FeedingPointDetails/Model/FeedingPointDetailsModel.swift

```diff
@@ -37,79 +42,35 @@ final class FeedingPointDetailsModel: FeedingPointDetailsModelProtocol, FeedingP
-                        completion?(
-                            self.mapper.map(
-                                point,
-                                isFavorite: self.relatedFavoritePoint != nil
-                            )
-                        )
-                    }
-                }
-            case .failure(let error):
-                // TODO: Handele error
-                print(error)
-            }
-        }
-    }
-
-    func mutateFavorite(completion: ((Bool) -> Void)?) {
-        guard let feedingPoint = feedingPoint, let userId = context.profileService.getCurrentUser()?.userId else {
-            completion?(false)
-            return
+    func fetchFeedingPoint(_ completion: ((FeedingPointDetailsModel.PointContent) -> Void)?) {
+        let fullFeedingPoint = context.feedingPointsService.storedfeedingPoints.first { point in
+            point.feedingPoint.id == self.feedingPointId
         }
 
-        var request: Request<Favourite>
-        if let favoritePoint = relatedFavoritePoint {
-            request = .delete(favoritePoint)
-        } else {
-            let favouritePoint = Favourite(
-                userId: userId,
-                feedingPointId: feedingPoint.id,
-                feedingPoint: feedingPoint
+        if let feedingPointModel = fullFeedingPoint {
+            self.fullFeedingPoint = feedingPointModel
+            completion?(
+                self.mapper.map(
+                    feedingPointModel.feedingPoint,
+                    isFavorite: feedingPointModel.isFavorite
+                )
             )
-            request = .create(favouritePoint)
-        }
-        context.networkService.mutate(request: request) { event in
-            switch event {
-            case .success:
-                DispatchQueue.main.async {
-                    completion?(true)
-                }
-            case .failure:
-                DispatchQueue.main.async {
-                    completion?(false)
-                }
-            }
         }
     }
 
-    private func fetchFavorites() async {
-        let userId = context.profileService.getCurrentUser()?.userId ?? ""
-        let fav = Favourite.keys
-        let predicate = fav.userId == userId
-        do {
-            let favorites = try await context.networkService.query(request: .list(Favourite.self, where: predicate))
-            if let identifier = self.feedingPoint?.id {
-                self.relatedFavoritePoint = favorites.first { fav in
-                    fav.feedingPoint.id == identifier
-                }
-            }
-        } catch {
-            logError(error.localizedDescription)
+    func mutateFavorite() async throws -> Bool {
+        guard let feedingPoint = fullFeedingPoint else {
+            return false
+        }
+        if feedingPoint.isFavorite {
+            try await context.feedingPointsService.deleteFromFavorites(byIdentifier: feedingPointId)
+            return true
```

Comment:
If in two cases we return true, it will be better to place `return true` after `if-else` construction.

Reply:
Fixed